src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -2203,6 +2207,48 @@ namespace AirflowNetwork {
                 ShowContinueError(m_state, "..Default value \"SkylineLU\" will be used.");
             }
 
+            // Get inputs for duct sizing
+            AirflowNetworkSimu.AFNDuctAutoSize = false;
+            if (NumAlphas == 10) {
+                if (!UtilityRoutines::SameString(Alphas(10), "None")) {
+                    if (UtilityRoutines::SameString(Alphas(10), "MaximumVelocity")) {
+                        AirflowNetworkSimu.ductSizeMethod = AirflowNetworkSimuProp::DuctSizeMethod::MaxVelocity;
+                    } else if (UtilityRoutines::SameString(Alphas(10), "PressureLoss")) {
+                        AirflowNetworkSimu.ductSizeMethod = AirflowNetworkSimuProp::DuctSizeMethod::PressureLoss;
+                    } else if (UtilityRoutines::SameString(Alphas(10), "PressureLossWithMaximumVelocity")) {
+                        AirflowNetworkSimu.ductSizeMethod = AirflowNetworkSimuProp::DuctSizeMethod::VelocityAndLoss;
+                    } else {
+                        ShowSevereError(m_state,
+                                        format(RoutineName) + CurrentModuleObject + " object, " + cAlphaFields(10) + " = " + Alphas(10) +
+                                            " is invalid.");
+                        ShowContinueError(m_state,
+                                          "Valid choices are None, MaximumVelocity, PressureLoss, and PressureLossWithMaximumVelocity. " +
+                                              CurrentModuleObject + ": " + cAlphaFields(1) + " = " + AirflowNetworkSimu.AirflowNetworkSimuName);
+                        ErrorsFound = true;
+                        SimObjectError = true;
+                    }
+                    AirflowNetworkSimu.AFNDuctAutoSize = true;
+                    if (SimulateAirflowNetwork != AirflowNetworkControlMultiADS) {
+                        ShowWarningError(m_state, format(RoutineName) + CurrentModuleObject + " object, ");
+                        ShowContinueError(m_state,
+                                          "Although " + cAlphaFields(10) + " = \"" + Alphas(10) + "\" is entered, but " + cAlphaFields(2) +
+                                              " is not MultizoneWithoutDistribution.");
+                        ShowContinueError(m_state, "..Duct sizing is not performed");
+                        AirflowNetworkSimu.AFNDuctAutoSize = false;
+                    }
+                    if (SimulateAirflowNetwork == AirflowNetworkControlMultiADS) {
```

Comment:
Should this not be just an `else`?

Reply:
Maybe so, but this can get resolved over with transition to an enum in #9449.

Reply:
I let @jasondegraw resolve them first.